tango_simlib/sim_xmi_parser.py

```diff
@@ -182,18 +182,35 @@ def __init__(self):
         }]
 
         """
-        self.class_description = []
-        """Data structure format is a list containing the Tango class description
-        information in a dict.
+        self.class_description = {}
+        """Data structure format is a dictionary containing the Tango class description
+        information.
 
         e.g.
-        [{
-           "super_class_num": {
+        {
+           "super_class_num": [{
                "classname": "",
                "sourcePath": "an absolute path to the parent xmi file."
+               }],
+           "description": "",
+           "title": "",
+           "sourcePath": "",
+           "language": "",
+           "filestogenerate": "",
+           "license": "",
+           "copyright": "",
+           "hasMandatoryProperty": "",
+           "hasConcreteProperty": "",
+           "hasAbstractCommand": "",
+           "hasAbstractAttribute": "",
+           "identification": {
+               "author": "",
+               "contact": "",
+               "emailDomain": ""
```

Comment:
In `extract_device_class_descr` you don't seem to be using putting all these attributes into `self.class_description`.